test/cluster/test_dedicated_rack_leaders.py

```
@@ -0,0 +1,262 @@
+
+        # Keep writing. A leader which replicates tells the followers who it is, so the
+        # preferred replica knows there is one and stays a follower even once it is allowed
+        # to campaign again - an idle group would instead get its leadership back for free,
+        # by that replica timing out first.
+        writing = True
+
+        async def writer():
+            pk = 0
+            while writing:
+                try:
+                    await cql.run_async(f"INSERT INTO {ks}.t (pk, v) VALUES ({pk}, {pk})")
```

Ruff (0.16.1)

[error] Possible SQL injection vector through string-based query construction

(S608)